linux/deploy_vm/upgrade_linux_kernel.yml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Take a snapshot before upgrading kernel"
+  include_tasks: ../../common/vm_take_snapshot.yml
+  vars:
+    snapshot_name: "kernel_before_upgrade"
+
+- name: "Get kernel version before upgrading"
+  ansible.builtin.set_fact:
+    kernel_is_upgraded: false
+    kernel_before_upgrade: "{{ guest_os_ansible_kernel }}"
+
+- name: "Upgrade Kernel for Oracle"
+  when: guest_os_ansible_distribution == "OracleLinux"
+  block:
+    - name: "Add Oracle Linux online repos for upgrading kernel"
+      include_tasks: ../utils/add_official_online_repo.yml
+    - name: "Update Oracle Linux 9.0 to latest UEK R7"
+      ansible.builtin.shell: "yum update --nogpgcheck -y"
+      register: kernel_upgrade_result
+      delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Update mainline kernel to {{ upgrade_kernel_version }} for Ubuntu"
+  ansible.builtin.shell: |
+    wget -c -- $(curl -s {{ mainline_kernel_url }}/v{{ upgrade_kernel_version }}/amd64/ | sed 's/"\|>\|</\n/g' | grep deb | uniq | while read line; do echo "{{ mainline_kernel_url }}/v{{ upgrade_kernel_version }}/amd64/${line}";done)
+    if [ $? -eq 0 ]; then dpkg -i linux*.deb; rm -rf linux*.deb; fi
+    sed -i 's/#WaylandEnable=false/WaylandEnable=false/'  /etc/gdm3/custom.conf
```

Comment:
This is not a good practice by having so many pipes in one shell command. It is not good at understanding. And it is lack of information for triaging issues. 

By the way for Oracle Linux this task's registration kernel_upgrade_result might overwrite the above one even though it has a when. You'd better to test such scenario with a simple playbook